scipy/stats/_distn_infrastructure.py

```diff
@@ -868,11 +871,31 @@ def _argcheck(self, *args):
             cond = logical_and(cond, (asarray(arg) > 0))
         return cond
 
-    def _support_mask(self, x):
-        return (self.a <= x) & (x <= self.b)
+    def _get_support(self, *args):
+        """Return the support of the (unscaled, unshifted) distribution.
+
+        *Must* be overridden by distributions which have support dependent
```

Comment:
Is the intent that when a new distribution is added and the distribution's support depends on a parameter value, that the value is set in the `_get_support` method that is overridden in the child class? 

If so maybe state that explicitly here? I read this as you must override the function with a distribution specific one, but *not* that the code in the `_get_support` should override the support limits.

If not, then where is the developer to place the overridden `a` and `b` parameters (for location and scale)

Reply:
It is not the intent to ever set `self.a` or `self.b` for such a distribution.  The issue is that a single instance of the class `truncexpon` is shared amongst all users of the `truncexpon ` distribution, so setting it in one circumstance sets it for all.  Even a frozen distribution is not actually frozen --- what are frozen are the constructor arguments so that it can use the distribution at some later time.

`self.a` and `self.b` do not need to get changed for location and scale --- handling that  happens automatically in the methods of `rv_continuous`,  such as `rv_continuous.cdf` etc.

Reply:
ok, so I understood correctly. I'd recommend adding a line here that states explicitly not to use `_get_support` to set the values either. I'm worried the message will not be explicit enough for future distribution creators. Maybe something like: 

"The `self.a` and `self.b` limits should not be set by this method, rather this is a utility method that returns the support limits for use in other distribution class methods. "

Or something similar if you prefer a different wording.

Reply:
I modified the docstring of `rv_generic` and `rv_continuous` to be more explicit about the need for overriding `_get_support`, and modified the docstring of `_get_support` to be explicit on not permitting any modification of any members of the class.

Reply:
Thanks for that, that will make it easier on new developers coming into the `scipy.stats` submodule in the future (and me when I need to remember this again).